core/src/test/scala/org/apache/spark/storage/ShuffleBlockFetcherIteratorSuite.scala

```diff
@@ -48,15 +49,22 @@ class ShuffleBlockFetcherIteratorSuite extends SparkFunSuite with PrivateMethodT
   // in the presence of faults.
 
   /** Creates a mock [[BlockTransferService]] that returns data from the given map. */
-  private def createMockTransfer(data: Map[BlockId, ManagedBuffer]): BlockTransferService = {
+  private def createMockTransfer(
+      data: Map[BlockId, ManagedBuffer],
+      throwNettyOOM: Boolean = false): BlockTransferService = {
     val transfer = mock(classOf[BlockTransferService])
+    var hasThrowOOM = false
     when(transfer.fetchBlocks(any(), any(), any(), any(), any(), any())).thenAnswer(
       (invocation: InvocationOnMock) => {
         val blocks = invocation.getArguments()(3).asInstanceOf[Array[String]]
         val listener = invocation.getArguments()(4).asInstanceOf[BlockFetchingListener]
+        val OOMBlockIndex = new Random(System.currentTimeMillis()).nextInt(blocks.length)
```

Comment:
I would avoid the random here but if you would like to keep this then let's log out this index to make the run more reproducible.

Reply:
The random intends to test both failure cases of "the last block of a request" and "a middle block of a request". I'll make it determined.